runtime/expr/function/grep.go

```diff
@@ -0,0 +1,43 @@
+package function
+
+import (
+	"fmt"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/runtime/expr"
+	"golang.org/x/text/unicode/norm"
+)
+
+type Grep struct {
+	grep   expr.Evaluator
+	grestr string
+	zctx   *zed.Context
+}
+
+func (g *Grep) Call(ectx zed.Allocator, vals []zed.Value) *zed.Value {
+	patternArg, inputArg := vals[0], vals[1]
+	if zed.TypeUnder(patternArg.Type) != zed.TypeString {
+		return g.error(ectx, "pattern argument must be a string", &patternArg)
+	}
+	if p := patternArg.AsString(); g.grep == nil || g.grestr != p {
+		g.grestr = p
+		term := norm.NFC.Bytes(patternArg.Bytes())
+		g.grep = expr.NewSearchString(string(term), nil)
+	}
+	return g.grep.Eval(wrapAllocator{ectx}, &inputArg)
+}
+
+// XXX This is gross, any reason function shouldn't just accept an expr.Context?
+type wrapAllocator struct {
+	zed.Allocator
+}
+
+func (wrapAllocator) Vars() []zed.Value { return nil }
+
+func (g *Grep) error(ectx zed.Allocator, err string, val *zed.Value) *zed.Value {
```

Comment:
Nit: `err` is for `error`s.
```suggestion
func (g *Grep) error(ectx zed.Allocator, msg string, val *zed.Value) *zed.Value {
```